### ISSUE TITLE
Twitch authentication (and thus Stream Information docks) from OBS 28 being ignored/overwritten

### ISSUE THREAD

**Issue by braezer** [pkviet/obs-studio#81]
activating the Twitch account in OBS 28.03 as per instructed to enable the docks

re-opening 28.03 confirms the settings have stuck (screens 1a&1b)
![1a](https://user-images.githubusercontent.com/116301237/197020747-cf9ffea5-2d7d-43ec-ac5c-bc0fead0aee2.PNG)
![1b](https://user-images.githubusercontent.com/116301237/197020749-0b5afca6-c45f-409c-996f-9838c344ee9c.PNG)

opened Music Edition to find docks are not loading (screens 2a&2b)
![2a](https://user-images.githubusercontent.com/116301237/197020750-63c75c58-4132-4c46-8d44-30ed11e01f7d.PNG)
![2b](https://user-images.githubusercontent.com/116301237/197020751-c18bf1d0-490e-47dc-8518-b6c4be711f54.PNG)

go back to 28 and find they're not loading there either & authentication has nulled yet stream-key is still recognised (screens 3a&3b)
![3a](https://user-images.githubusercontent.com/116301237/197020752-ca514df6-c0df-483b-93bf-15a989590b95.PNG)
![3b](https://user-images.githubusercontent.com/116301237/197020754-aa977006-decd-4252-a5da-0c227e1434e1.PNG)

checked the logs, noticed websocket isn't included with Music Edition so installed the v27 compatible version to ME- no effect
noticed Qt & obs-browser differences - not touched in case it further breaks things and because I can't find obs-broswer 2.18.5

tried with YouTube, chat's enabled and swaps over to both with no problem, however no way to manage stream e.g. editing title etc. as twitch can (and honestly, I forget if I did this step first, or did it after doing Twitch, and if the latter I can't remember if it worked) (screens 4a&4b)
![4a](https://user-images.githubusercontent.com/116301237/197020757-71b332df-62ad-49f1-b2d7-b7b92b304ba4.PNG)
![4b](https://user-images.githubusercontent.com/116301237/197020760-2b3ae028-4133-402b-99ac-a3d807a83e3d.PNG)

tried to see what differences there were in logs when swapping between OBS version, including launching them simultaneously once 28 was confirmed working - no effect

tried with different Twitch Servers instead of the Auto setting - no effect

-----------------------------
what I found was:
inside global.ini, twitch authentication stuff was still there after I went into ME
going back into 28 I re-authenticated and noticed there was only one difference and it was that ME is clearing this portion:

[Auth]
Type=Twitch

and, whilst keeping the [Auth] header, there was nothing underneath. when doing the youtube authentication (which works from being set in ME natively), it would say Type=Youtube      -     RTMP (or something to that effect)
so I set global.ini as Read-Only with Type=Twitch in place, but ME was flat-out ignoring it, however on re-launching 28, it was still activated and the Stream Information gubbins was still active

additionally, being a UK Denizen, I noticed that in ME there is only one option for "EU: London", however in 28 there are 4 servers consisting of "Europe: London" each with a suffix of (3), (4), (8), and (8) ((yes, there were TWO 8's)) that give different server URLs when swapped to ME. trying different pertubations of server selection didn't work
----------------------------

any idea on what's happening? if you need any more info I can supply it if necessary. and thanks in advance
apart from installing the websocket plugin, the only other thing I've added is "obs-multi-rtmp" with the intent to simultaneously stream to YouTube and Twitch

**Comment by braezer** [comment]
just found the below, it's not related is it? I.E. a change in the Twitch API that ME needs to reconcile? Can't seem to see a mention on which version of the Twith API is being used for ME, and on the project's latest release (July 2022 v27.2.5) under Features it mentions 
     "It is based on OBS-Studio v27.0.1 and includes all its features." 
which contradicts higher up where it says 
     "Apart from that it should be identical to vanilla obs 27.2.4."
----------------------------------------------------
In a post on Feb. 15, OBS revealed that with the release of OBS Studio version 27.2 the previous versions of the software will be made redundant on Monday, Feb. 28.

    🚨 [Twitch](https://geni.us/ybEcleV) Users 🚨 You MUST update to 27.2 before the end of February! After Feb 28th, you will no longer be able to connect your Twitch account in any previous version of OBS due to Twitch ending support for the old v5 API. Only OBS Studio 27.2 and later support the new API.
    — OBS (@OBSProject) [February 14, 2022]


P.S. changed the title to better highlight the issue at a first glance

**Comment by pkviet** [comment]
The build is definitely based on 27.2.4.
I'll try to update it and fix the issues for 28.1 or in all likeliness for 29.
I have scant time to support this fork which is totally on my free time.
As to 3rd party plugins, they can't work on ME because they have to be compiled against my libobs which differs from vanilla libobs.

**Comment by braezer** [comment]
Fair enough, thanks for your time and all the effort you're putting/put in, hopefully my issue is a simple fix that won't take up too much of your time for such a minor inconvenience.

fingers crossed OBS recognises this fork and adds in the features you and others have devloped natively to ease the burden on yourself. honestly don't know why the features you've brought to the table AREN'T in the main build (my only guess is YouTube's lack-luster audio capabilities compared to Twitch but that's just speculation)
